src/main/java/skynet/Factory.java

```diff
@@ -0,0 +1,47 @@
+package skynet;
+
+import java.util.*;
+
+/**
+ * Represents a neutral factory that produces robot parts.
+ * Parts are produced randomly, and factions collect them each night.
+ */
+public class Factory {
+    private final List<Part> parts = new ArrayList<>();
+
+    /**
+     * Produces up to 10 random parts and adds them to the inventory.
+     *
+     * @param random the random generator used for producing parts
+     */
+    public synchronized void produceParts(Random random) {
```

Comment:
While methods are synchronized, there's no proper coordination between threads to ensure the day/night cycle is respected.